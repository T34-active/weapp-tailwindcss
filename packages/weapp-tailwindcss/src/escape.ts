export const weappTwIgnore = String.raw

export { replaceWxml as escape } from './wxml/shared'

export { isAllowedClassName } from '@weapp-core/escape'
