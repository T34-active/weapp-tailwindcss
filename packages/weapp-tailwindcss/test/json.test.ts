import { isAllowedClassName } from '@weapp-core/escape'

describe('json', () => {
  it('class allow', async () => {
    const res = await import('./fixtures/json/any.json')
    const a = []
    const b = []
    for (const x of res.default) {
      if (isAllowedClassName(x)) {
        a.push(x)
      }
      else {
        b.push(x)
      }
    }
    expect(a).toMatchSnapshot('Allowed')
    expect(b).toMatchSnapshot('Not Allowed')
  })
})
