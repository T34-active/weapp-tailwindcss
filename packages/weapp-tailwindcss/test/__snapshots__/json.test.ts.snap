// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`json > class allow > Allowed 1`] = `
[
  "-mt-2",
  "bg-blue-200",
  "bg-blue-400",
  "bg-gradient-to-b",
  "bg-green-200",
  "bg-green-400",
  "bg-pink-500",
  "bg-red-900",
  "block",
  "border-4",
  "border-blue-400",
  "border-green-400",
  "border-solid",
  "box-border",
  "box-content",
  "container",
  "contents",
  "decoration-clone",
  "divide-green-500",
  "divide-solid",
  "divide-x",
  "flex",
  "float-left",
  "float-none",
  "float-right",
  "flow-root",
  "font-semibold",
  "from-yellow-400",
  "grid",
  "grid-cols-3",
  "h-32",
  "h-5",
  "h-full",
  "hidden",
  "inline",
  "inline-block",
  "inline-flex",
  "inline-grid",
  "inline-table",
  "items-center",
  "justify-center",
  "p-4",
  "py-2",
  "ring-4",
  "ring-pink-300",
  "rounded-md",
  "shadow",
  "shadow-2xl",
  "table",
  "table-caption",
  "table-cell",
  "table-column",
  "table-column-group",
  "table-footer-group",
  "table-header-group",
  "table-row",
  "table-row-group",
  "text-black",
  "text-sm",
  "text-transparent",
  "text-white",
  "to-red-500",
  "w-32",
  "w-5",
  "w-full",
]
`;

exports[`json > class allow > Not Allowed 1`] = `
[
  "!h-5",
  "-m-[20px]",
  "-wx:bg-red-500",
  "after:content-['这是classArray,用于测试动态js绑定class']",
  "before:block",
  "before:content-['hello']",
  "bg-[#000000]",
  "bg-[#00ff00]",
  "bg-[#123456]",
  "bg-[#dddddd]",
  "bg-[#fafa00]",
  "bg-[#ff0000]",
  "bg-[url('https://xxx.com/xx.webp')]",
  "bg-opacity-[0.54]",
  "border-[#098765]",
  "border-[10px]",
  "border-opacity-[0.44]",
  "bottom-[100px]",
  "bottom-[100rpx]",
  "divide-[#010101]",
  "divide-x-[10px]",
  "h-[100px]",
  "h-[200%]",
  "ifdef-[MP-WEIXIN]:bg-blue-500",
  "ifndef-[MP-WEIXIN]:bg-red-500",
  "leading-[0.9]",
  "max-h-[100px]",
  "max-w-[300rpx]",
  "mb-[-20px]",
  "min-h-[100px]",
  "min-w-[300rpx]",
  "p-[20px]",
  "rounded-[40px]",
  "shadow-[0px_2px_11px_0px_#00000a]",
  "shadow-[0px_2px_11px_0px_rgba(0,0,0,0.4)]",
  "space-y-[1.6rem]",
  "text-[#132323]",
  "text-[#898989]",
  "text-[#dddddd]",
  "text-[#fafafa]",
  "text-[#ffffff]",
  "text-[20px]",
  "text-[22px]",
  "text-[30rpx]",
  "text-opacity-[0.19]",
  "w-[100px]",
  "w-[100rpx]",
  "w-[300rpx]",
  "wx:bg-blue-500",
  "z-[-55]",
  "z-[55]",
]
`;
